linuxHelpers.cpp

```diff
@@ -855,6 +856,128 @@ VOID LargeIntegerToSystemTimeString(
+
+    hashFlag[ALGO_SHA1] = (((hashType>>(ALGO_SHA1-1))&1) && (hashType & ALGO_MULTIPLE)) || (hashType == ALGO_SHA1);
+    hashFlag[ALGO_MD5] = (((hashType>>(ALGO_MD5-1))&1) && (hashType & ALGO_MULTIPLE)) || (hashType == ALGO_MD5);
+    hashFlag[ALGO_SHA256] = (((hashType>>(ALGO_SHA256-1))&1) && (hashType & ALGO_MULTIPLE)) || (hashType == ALGO_SHA256);
+
+    filePtr = fopen(imagePath, "rb");
+    if( !filePtr ) return;
+
+    // Read and hash image
+    while((n = fread(tmpReadBuffer, 1, sizeof(tmpReadBuffer), filePtr))){
+
+        if( hashFlag[ALGO_SHA1] ){
+            if (!EVP_DigestUpdate(sha1_ctx, tmpReadBuffer, n)){
+                fclose(filePtr);
+                return;
+            }
+            if ( !(hashType & ALGO_MULTIPLE) ) continue;
+        }
+
+        if( hashFlag[ALGO_MD5] ){
+            if (!EVP_DigestUpdate(md5_ctx, tmpReadBuffer, n)){
+                fclose(filePtr);
+                return;
+            }
+            if ( !(hashType & ALGO_MULTIPLE) ) continue;
+        }
+
+        if( hashFlag[ALGO_SHA256] ){
+            if (!EVP_DigestUpdate(sha256_ctx, tmpReadBuffer, n)){
+                fclose(filePtr);
+                return;
+            }
+            if ( !(hashType & ALGO_MULTIPLE) ) continue;
+        }
+
+    }
+    fclose(filePtr);
+
+    // Retrieve digest and cleanup
+    EVP_DigestFinal(sha1_ctx  , tmpHashBuffer[ALGO_SHA1]  , &hashSize[ALGO_SHA1]);
+    EVP_DigestFinal(md5_ctx   , tmpHashBuffer[ALGO_MD5]   , &hashSize[ALGO_MD5]);
+    EVP_DigestFinal(sha256_ctx, tmpHashBuffer[ALGO_SHA256], &hashSize[ALGO_SHA256]);
+
+    memset(tmpStringBuffer, 0, sizeof(tmpStringBuffer));
+    memset(tmpHashPrefixBuffer, 0, sizeof(tmpHashPrefixBuffer));
+
+    for(unsigned int algo=0;algo<ALGO_MAX;algo++){
+        if( hashFlag[algo] ){
+
+            // Add seperator if multiple hashes
+            if( *tmpStringBuffer ){
+                memset(tmpStringBuffer, 0, sizeof(tmpStringBuffer));
+                memset(tmpHashPrefixBuffer, 0, sizeof(tmpHashPrefixBuffer));
+                
+                strcat((char *)tmpHashPrefixBuffer, hashSeparator);
+            }
+
+            // Add hash prefix "SHA1=", "MD5=" or "SHA256="
+            switch(algo){
+                case ALGO_SHA1:
+                    strcat((char *)tmpHashPrefixBuffer, hashPrefix[ALGO_SHA1-1]);
+                    break;
+                case ALGO_MD5:
+                    strcat((char *)tmpHashPrefixBuffer, hashPrefix[ALGO_MD5-1]);
+                    break;
+                case ALGO_SHA256:
+                    strcat((char *)tmpHashPrefixBuffer, hashPrefix[ALGO_SHA256-1]);
+                    break;
+            }
+
+            // Digest to hex
+            for(unsigned int i=0;i<hashSize[algo];i++){
+                snprintf((char *)tmpStringBuffer+i*2, sizeof(tmpStringBuffer), "%02x", tmpHashBuffer[algo][i]);
+            }
+
+            strcat(stringBuffer, (char *)tmpHashPrefixBuffer);
```

Comment:
I think stringBuffer is [256] so it might overflow here with both a tmpHashPrefixBuffer and tmpStringBuffer.